City-of-Helsinki/hel-fi-drupal-grants
# Feature: AU-1523: Implemented year validation for custom composite elements

# [AU-1523](https://helsinkisolutionoffice.atlassian.net/browse/AU-1523)

## What was done
This pull request implements validation for the year field in the `myonnetty_avustus` and `haettu_avustus_tieto` custom composite Webform elements. Now a user can only enter a year between 1900 - 2100, where as before any 4-digit number would have worked.

## How to install
Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-1523-form-year-checks`
  * `make fresh`
  * `make drush-cr`

## How to test
- Import the form configurations with `drush gwi`.
- Open of up the forms with `drush gwco`.
- Test any form with either a `myonnetty_avustus` or `haettu_avustus_tieto` field (make sure that the form is not config ignored). You should only be able to enter a year between 1900 - 2100.

[AU-1523]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1523?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
